pyanaconda/ui/helpers.py

```diff
@@ -76,36 +75,24 @@ class SourceSwitchHandler(object, metaclass=ABCMeta):
     """
 
     @abstractproperty
-    def data(self):
-        pass
-
-    @abstractproperty
-    def storage(self):
+    def payload(self):
         pass
 
     def __init__(self):
         self._device = None
         self._current_iso_path = None
 
-    def unset_source(self):
-        """Unset an already selected source method.
-
-        Unset the source in kickstart and notify the payload so that it can correctly
-        release all related resources (unmount iso files, drop caches, etc.).
-        """
-        self._clean_hdd_iso()
-        self.data.method.method = None
-        payloadMgr.restart_thread(self.payload, checkmount=False)   # pylint: disable=no-member
-        threadMgr.wait(constants.THREAD_PAYLOAD_RESTART)
-        threadMgr.wait(constants.THREAD_PAYLOAD)
-
-    def _clean_hdd_iso(self):
+    def _tear_down_existing_source(self):
         """ Clean HDD ISO usage
         This means unmounting the partition and unprotecting it,
         so it can be used for the installation.
         """
```

Comment:
The docstring no longer matches, I think. The cleaning feels like side effect...

Reply:
True, thanks!